### DISCUSSION TITLE
Applying obtained deformation fields on segmentation masks that correspond to the moving part.

### DISCUSSION THREAD

**Discussion by Kevin-Miao** [voxelmorph/voxelmorph#397] [Q&A]
Hi!

I was wondering if anyone had any pointers on how to apply the obtained deformation fields to segmentation mask that correspond to the moving image?

**Comment by Sma6500** [comment]
Hi ! If I'm wright you would like to registered the segmentation mask from the moving image on your fixed image ? To achieve that in pytorch you just need to apply the transformer layer to the mask with your deformation fields :
moved_mask = net.transformer(moving_image_mask.float(), flow_field)
It should be the same in tensorflow.

**Reply by simonaturco** [reply]
Hi,
how can you do the same with tensorflow? Is 'net' my trained model?

**Reply by adalca** [reply]
@simonaturco did you figure it out? We actually enable this directly into the model with model.apply_transform()
https://github.com/voxelmorph/voxelmorph/blob/7cf611ab283b22424d82650d6e17d49b986e7981/voxelmorph/tf/networks.py#L277

**Comment by Kevin-Miao** [comment]
Hey Sma! Thanks for the response! :) Didn't realize it was the same procedure in torch as tf.
